Code/Model.py

```diff
@@ -0,0 +1,174 @@
+        """
+
+        model = sk.svm.SVC(C, kernel, degree, gamma, coef0, max_iter=max_iter)
+        super().__init__({'C':C, 'kernel':kernel, 'degree':degree, 'gamma':gamma, 'coef0':coef0})
+
+
+    def train(self, X_train, t_train):
+
+        """
+        Train our model
+
+        :param X_train: NxD numpy array of the observations of the training set {N : nb of obs, D : nb of dimensions}
+        :param t_train: Nx1 numpy array of classes associated with each observation in the training set
+        """
+
+        self.model.fit(X_train, t_train)
+
+
+    def predict(self, X):
+
+        """
+        Predict classes for our observations in the input array X
+
+        :param X: NxD numpy array of observations {N : nb of obs, D : nb of dimensions}
+        :return: Nx1 numpy array of classes predicted for each observation
+
+        """
+
+        return self.model.predict(X)
+
+
+class MLP(Model):
+
+    def __init__(self, hidden_layer_sizes=(100, ), activation='relu', solver='adam', alpha=0.0001, batch_size='auto',
+                 learning_rate='constant', learning_rate_init=0.001, power_t=0.5, max_iter=200, momentum=0.9, beta_1=0.9,
+                 beta_2=0.999):
+
+        """
+
+        Class that generate a Multi-layer Perceptron classifier.
+
+        This model optimizes the log-loss function using LBFGS or stochastic gradient descent.
+
+        Some hyperparameters are conditional to others!
+        For more informations take a look at :
+        https://scikit-learn.org/stable/modules/generated/sklearn.neural_network.MLPClassifier.html#sklearn.neural_network.MLPClassifier
+
+        :param hidden_layer_sizes: The ith element represents the number of neurons in the ith hidden layer
+        :param activation: Activation function for the hidden layer {‘identity’, ‘logistic’, ‘tanh’, ‘relu’}
+        :param solver: The solver for weight optimization {‘lbfgs’, ‘sgd’, ‘adam’}
+        :param alpha: L2 penalty (regularization term) parameter
+        :param batch_size: Size of minibatches for stochastic optimizers.
+        :param learning_rate: Learning rate schedule for weight updates {‘constant’, ‘invscaling’, ‘adaptive’}
+        :param learning_rate_init: The initial learning rate used
+        :param power_t: The exponent for inverse scaling learning rate.
+        :param max_iter: Maximum number of iterations
+        :param momentum: Momentum for gradient descent update
+        :param beta_1: Exponential decay rate for estimates of first moment vector in adam, should be in [0, 1)
+        :param beta_2: Exponential decay rate for estimates of second moment vector in adam, should be in [0, 1)
+        """
+
+        model = sk.neural_network.MLPClassifier(hidden_layer_sizes=(100, ), activation='relu', solver='adam',
+                                                alpha=0.0001, batch_size='auto',learning_rate='constant', learning_rate_init=0.001,
```

Comment:
valeurs hardcodées?

Reply:
It's a mistake. Thank you! No such simple mistake will be made in future.